perf_bench_test.go

```diff
@@ -24,127 +24,567 @@ package main
 
 import (
 	"context"
+	"encoding/csv"
 	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
 	"strconv"
+	"strings"
 	"sync"
+	"sync/atomic"
 	"testing"
 	"time"
 
+	"github.com/ease-lab/vhive/metrics"
+	"github.com/ease-lab/vhive/profile"
+	"github.com/montanaflynn/stats"
 	log "github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/require"
 )
 
 var (
-	isColdStart     = flag.Bool("coldStart", false, "Profile cold starts (default is false)")
-	vmNum           = flag.Int("vm", 2, "The number of VMs")
-	targetReqPerSec = flag.Int("requestPerSec", 2, "The target number of requests per second")
-	executionTime   = flag.Int("executionTime", 2, "The execution time of the benchmark in seconds")
+	// shared arguments
+	warmUpTime   = flag.Float64("warmUpTime", 5, "The warm up time before profiling in seconds")
+	coolDownTime = flag.Float64("coolDownTime", 1, "The cool down time after profiling in seconds")
+	profileTime  = flag.Float64("profileTime", 10, "The profiling time in seconds")
+	funcNames    = flag.String("funcNames", "helloworld", "Names of the functions to benchmark, separated by comma")
+	isBindSocket = flag.Bool("bindSocket", false, "Bind all VMs to socket 1")
+
+	vmNum     = flag.Int("vm", 2, "TestProfileSingleConfiguration: The number of VMs")
+	targetRPS = flag.Int("rps", 10, "TestProfileSingleConfiguration: The target requests per second")
+
+	vmIncrStep = flag.Int("vmIncrStep", 4, "TestProfileIncrementConfiguration: The increment VM number")
+	maxVMNum   = flag.Int("maxVMNum", 100, "TestProfileIncrementConfiguration: The maximum VM number")
+
+	// profiler arguments
+	profilerLevel    = flag.Int("l", 1, "Profile level (-l flag)")
+	profilerInterval = flag.Uint64("I", 500, "Print count deltas every N milliseconds (-I flag)")
+	profilerNodes    = flag.String("nodes", "", "Include or exclude nodes (with "+
+		"+ to add, "+
+		"-|^ to remove, "+
+		"comma separated list, wildcards allowed, "+
+		"add * to include all children/siblings, "+
+		"add /level to specify highest level node to match, "+
+		"add ^ to match related siblings and metrics, "+
+		"start with ! to only include specified nodes)")
 )
 
-func TestBenchRequestPerSecond(t *testing.T) {
+func TestProfileIncrementConfiguration(t *testing.T) {
```

Comment:
please add a comment, describing what kind of experiment this test runs

Reply:
Done